tests/tests.rs

```diff
@@ -95,5 +127,12 @@ mod tests {
                 prop_assert_approx_eq!(v, expected);
             }
         }
+
+        #[test]
+        fn rms_square_wave(f in arb_frequency(), p in arb_timesecs()) {
+            prop_assume!(f > FrequencyHz(0));
+            prop_assume!(p > TimeSecs(0.0));
+            prop_assert_approx_eq!(rms(square_wave(f), p), 1.0);
```

Comment:
I tried translating the pseudocode to Rust naively, thinking it could be useful as a test oracle. Oddly, the tests fail when I replace `rms` with the following:

```rust
/// Estimates the root mean square (RMS) of a signal over a period
///
/// # Arguments
///
/// * `s` - The `Signal` used to compute the RMS
/// * `period` - The period (starting at phase=0) over which the computation is made
///
pub fn rms(s: Signal, period: TimeSecs) -> f64 {
    // The number of samples should depend on the period.
    let samples = 100;
    let rate = FrequencyHz(((samples as f64) / period.0) as u32);
    let a: Vec<_> = sample(rate, s).map(|x| x.powf(2.0)).take(samples).collect();

    fn mean(a: Vec<f64>, n: usize) -> f64 {
        if n == 0 {
            0.0
        }
        else {
            let a_n = a[n];
            let mean_n_minus_1 = mean(a, n - 1);
            mean_n_minus_1 + (a_n - mean_n_minus_1) / ((n as i32) as f64)
        }
    }
    let n = a.len() - 1;
    mean(a, n).sqrt()
}
```

Just an observation, not really actionable.

Reply:
I had a look but I can't see what's going wrong here.